main/src/com/google/refine/operations/row/RowKeepMatchedOperation.java

```diff
@@ -0,0 +1,72 @@
+
+package com.google.refine.operations.row;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import com.google.refine.browsing.Engine;
+import com.google.refine.browsing.EngineConfig;
+import com.google.refine.browsing.FilteredRows;
+import com.google.refine.browsing.RowVisitor;
+import com.google.refine.history.HistoryEntry;
+import com.google.refine.model.Project;
+import com.google.refine.model.Row;
+import com.google.refine.model.changes.RowRemovalChange;
+import com.google.refine.operations.EngineDependentOperation;
+
+public class RowKeepMatchedOperation extends EngineDependentOperation {
+
+    @JsonCreator
+    public RowKeepMatchedOperation(
+            @JsonProperty("engineConfig") EngineConfig engineConfig) {
+        super(engineConfig);
+    }
+
+    @Override
+    protected String getBriefDescription(Project project) {
+        return "Keep only matching rows";
```

Comment:
This could be localized (especially since you have already introduced the `row_keep_matching_brief` message for it).